enclave_build/src/lib.rs

```diff
@@ -12,14 +12,71 @@ mod yaml_generator;
 use aws_nitro_enclaves_image_format::defs::{EifBuildInfo, EifIdentityInfo, EIF_HDR_ARCH_ARM64};
 use aws_nitro_enclaves_image_format::utils::identity::parse_custom_metadata;
 use aws_nitro_enclaves_image_format::utils::{EifBuilder, SignEnclaveInfo};
-use docker::DockerUtil;
+use docker::{DockerError, DockerUtil};
 use serde_json::json;
 use sha2::Digest;
 use std::collections::BTreeMap;
-use yaml_generator::YamlGenerator;
+use thiserror::Error;
+use yaml_generator::{YamlGenerator, YamlGeneratorError};
 
 pub const DEFAULT_TAG: &str = "1.0";
 
+#[derive(Debug, Error)]
+pub enum EnclaveBuildError {
+    #[error("Docker error: `{0:?}`")]
+    DockerError(DockerError),
+    #[error("Invalid path: `{0}`")]
+    PathError(String),
+    #[error("Image pull error: `{0}`")]
+    ImagePullError(String),
+    #[error("Image inspect failed")]
+    ImageInspectError,
+    #[error("Linuxkit error: `{0}`")]
+    LinuxKitError(String),
+    #[error("File operation error: `{0:?}`")]
+    FileError(std::io::Error),
+    #[error("Ramfs error: `{0:?}`")]
+    RamfsError(YamlGeneratorError),
+    #[error("Signature error: `{0}`")]
+    SignError(String),
+    #[error("Metadata error: `{0}`")]
+    MetadataError(String),
+    #[error("Unsupported architecture")]
+    UnsupportedArchError,
+    #[error("Getting image detail failed: `{0}`")]
+    ImageDetailError(String),
+    #[error("Container image operation failed")]
+    ImageOperationError,
+    #[error("Failed to convert image to EIF")]
+    ImageConvertError,
+    #[error("Hashing error: `{0}`")]
+    HashingError(String),
+    #[error("Serde error: `{0:?}`")]
+    SerdeError(serde_json::Error),
+    #[error("Error while parsing credentials: `{0}`")]
+    CredentialsError(String),
+    #[error("Image cache initialization error: `{0:?}`")]
+    CacheInitError(std::io::Error),
```

Comment:
There are some variants here that will be used in subsequent commits. Ideally we should add them at that point, but if that takes too much time, please mention in the commit message about this thing.